opencog/nlp/relex2logic/det.scm

```diff
@@ -0,0 +1,59 @@
+
+(define det
+    (BindLink
+        (VariableList
+            (TypedVariableLink
+                (VariableNode "$a-parse")
+                (TypeNode "ParseNode")
+            )
+            (TypedVariableLink
+                (VariableNode "$noun")
+                (TypeNode "WordInstanceNode")
+            )
+            (TypedVariableLink
+                (VariableNode "$det")
+                (TypeNode "WordInstanceNode")
+            )
+        )
+        (AndLink
+            (WordInstanceLink
+                (VariableNode "$noun")
+                (VariableNode "$a-parse")
+            )
+            (WordInstanceLink
+                (VariableNode "$det")
+                (VariableNode "$a-parse")
+            )
+            (EvaluationLink
+                (DefinedLinguisticRelationshipNode "_det")
+                (ListLink
+                    (VariableNode "$noun")
+                    (VariableNode "$det")
+                )
+            )
+            (InheritanceLink 
+                (VariableNode "$noun")
```

Comment:
@williampma there is an `InheritanceLink`, and (justed check with @misgeatgit ) the weights associated with the rules have no effect in giving priority to the application of the rules in the first and only iteration. Thus, no guarantee that this rule wil be run after the rest of the rules are applied, in the first iteration. In addition, the number of iterations has no effect as there is only a single iteration.